src/test/controllerscriptenginelegacy_test.cpp

```diff
@@ -693,13 +690,8 @@ TEST_F(ControllerScriptEngineLegacyTest, screenWontSentRawDataIfNotConfigured) {
             "Could not find a valid transform function but the screen doesn't "
             "accept raw data. Aborting screen rendering.");
 
-    transformScreenFrameFunctions().insert(
-            dummyScreen.identifier,
-            newTransformScreenFrameFunction(
-                    QMetaMethod(),
-                    false));
     renderingScreens().insert(dummyScreen.identifier, pDummyRender);
-    rootItems().insert(dummyScreen.identifier, std::make_shared<QQuickItem>());
+    rootItems().emplace(dummyScreen.identifier, std::make_unique<mixxx::qml::QmlMixxxControllerScreen>());
```

Comment:
same rationale for `push_back` vs `emplace_back` on `std::vector<std::unique_ptr<T>>` here:
```suggestion
    rootItems().insert(dummyScreen.identifier, std::make_unique<mixxx::qml::QmlMixxxControllerScreen>());
```

Reply:
What is the gain here? Also, please not that this is a test file so macro optimisation is unnecessary.

Reply:
Ah good to know! I thought `unique_ptr` were pretty safe when it come to leakge protection, but great to learn they can still s**** you :sweat_smile:  
In the context of unittest, a memory allocation failure would fail the test. Since that one-liner suggestion doesn't work, are you happy with keeping this as-is to limit the effort?

Reply:
ah yeah, I think my knowledge on `unordered_map` is not quite good enough to make an educated decision here. Keep as is for now.